From 3c14b0674b294eacf82fd2465c7b7d5fb55562c3 Mon Sep 17 00:00:00 2001
From: Guo Yi <yguo@cavium.com>
Date: Fri, 18 Dec 2020 00:49:03 +0000
Subject: [PATCH 1452/1921] spi: orion: Cap maximum speed of spi clock in cp11x
 to 41MHz

The SPI controller in cp11x only has a maximum speed of 41MHz
a new parameter block is added to distinguish the controller type

Change-Id: I5bd9960bb6ea8a00c915cabae3984c5e03aa438e
Signed-off-by: Guo Yi <yguo@cavium.com>
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/50161
Tested-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-cp11x.dtsi |  4 ++--
 drivers/spi/spi-orion.c                       | 11 +++++++++++
 2 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
index 6d013703c6e4..cf8be27e128c 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
@@ -382,7 +382,7 @@
 		};
 
 		CP11X_LABEL(spi0): spi@700600 {
-			compatible = "marvell,armada-380-spi";
+			compatible = "marvell,armada-cp110-spi";
 			reg = <0x700600 0x50>;
 			#address-cells = <0x1>;
 			#size-cells = <0x0>;
@@ -393,7 +393,7 @@
 		};
 
 		CP11X_LABEL(spi1): spi@700680 {
-			compatible = "marvell,armada-380-spi";
+			compatible = "marvell,armada-cp110-spi";
 			reg = <0x700680 0x50>;
 			#address-cells = <1>;
 			#size-cells = <0>;
diff --git a/drivers/spi/spi-orion.c b/drivers/spi/spi-orion.c
index 6643ccdc2508..46d6d20744bc 100644
--- a/drivers/spi/spi-orion.c
+++ b/drivers/spi/spi-orion.c
@@ -559,6 +559,13 @@ static const struct orion_spi_dev armada_380_spi_dev_data = {
 	.is_errata_50mhz_ac = true,
 };
 
+static const struct orion_spi_dev armada_cp110_spi_dev_data = {
+	.typ = ARMADA_SPI,
+	.max_hz = 41000000,
+	.max_divisor = 1920,
+	.prescale_mask = ARMADA_SPI_CLK_PRESCALE_MASK,
+};
+
 static const struct of_device_id orion_spi_of_match_table[] = {
 	{
 		.compatible = "marvell,orion-spi",
@@ -580,6 +587,10 @@ static const struct of_device_id orion_spi_of_match_table[] = {
 		.compatible = "marvell,armada-390-spi",
 		.data = &armada_xp_spi_dev_data,
 	},
+	{
+		.compatible = "marvell,armada-cp110-spi",
+		.data = &armada_cp110_spi_dev_data,
+	},
 	{
 		.compatible = "marvell,armada-xp-spi",
 		.data = &armada_xp_spi_dev_data,
-- 
2.31.1

